component/config.go

```diff
@@ -110,6 +112,47 @@ func callValidateIfPossible(v reflect.Value) error {
 // Type is the component type as it is used in the config.
 type Type string
 
+var _ fmt.Stringer = (Type)("")
```

Comment:
nit: did @bogdandrutu want to start moving these to the test files? This file already needs to use `fmt` so IDK if there is a benefit to moving it.

Reply:
See https://github.com/open-telemetry/opentelemetry-collector/issues/9431

Reply:
Moved!